core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```diff
@@ -62,16 +62,27 @@ public static String toJson(Expression expression, boolean pretty) {
     return JsonUtil.generate(gen -> toJson(expression, gen), pretty);
   }
 
+  public static String toJson(Expression expression, boolean pretty, boolean includeFieldIds) {
```

Comment:
Have we looked at if we should just have another API instead of passing in a boolean flag? Or instead of a boolean some kind of `ExpressionSerializationMode` enum so it's a bit more flexible